src/main/java/techcourse/myblog/domain/ArticleRepository.java

```diff
@@ -1,15 +1,54 @@
 package techcourse.myblog.domain;
 
 import org.springframework.stereotype.Repository;
+import techcourse.myblog.web.exception.NotExistEntityException;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
 
 @Repository
 public class ArticleRepository {
-    private List<Article> articles = new ArrayList<>();
+    private int newArticleId = 0;
+
+    private Map<Integer, Article> articles = new TreeMap<>();
 
     public List<Article> findAll() {
-        return articles;
+        return new ArrayList<>(articles.values());
+    }
+
+    public int insertArticle(Article article) {
+        int id = newArticleId++;
```

Comment:
`AUTO_INCREMENT` 👍 `id`가 0부터 시작하는 것이 맞는지 확인 부탁해요.

```suggestion
        int id = newArticleId + 1;
```